client/gutenberg/extensions/mailchimp/view.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * Internal dependencies
+ */
+import './view.scss';
+
+const blockClassName = 'wp-block-jetpack-mailchimp';
+
+const JetpackEmailSubscribe = {
+	validateEmail: function( email ) {
+		const re = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+		return re.test( String( email ).toLowerCase() );
+	},
+	fetch: ( blogId, email ) => {
+		const url =
+			'https://public-api.wordpress.com/rest/v1.1/sites/' +
+			encodeURI( blogId ) +
```

Comment:
I believe we need `encodeURIComponent`:

```js
url.parse( 'a/b/dirty?path/c' ).pathname
// 💥 'a/b/dirty'
url.parse( 'a/b/clean%3Fpath/c' ).pathname
// 😎 'a/b/clean%3Fpath/c'
```
```suggestion
			encodeURIComponent( blogId ) +
```

Reply:
Yes, agreed.

Reply:
34309305c5a9e7b7887ad05721da96fb4c4dfde4